src/dug/core.py

```diff
@@ -576,7 +587,22 @@ def expand_concept(self, concept):
                     concept.add_kg_answer(answer, query_name=query_name)
 
 
-def get_parser(parser_type):
+def get_parser(hook, parser_name):
+    """Get the parser from all parsers registered via the define_parsers hook"""
+
+    available_parsers = {}
+    hook.define_parsers(parser_dict=available_parsers)
+    parser = available_parsers.get(parser_name.lower())
+    if parser is not None:
+        return parser
+
+    err_msg = f"Cannot find parser of type '{parser_name}'\n" \
+              f"Supported parsers: {', '.join(available_parsers.keys())}"
+    logger.error(err_msg)
+    raise ParserNotFoundException(err_msg)
+
+
+def get_parser_old(parser_type):
```

Comment:
Is there a reason for keeping this around?

Reply:
No, that was just some old code that should be removed